test/Sanitizers/tsan/once.swift

```diff
@@ -0,0 +1,50 @@
+// RUN: %target-swiftc_driver %s -Xfrontend -parse-as-library -g -sanitize=thread %import-libdispatch -o %t_tsan-binary
+// RUN: %target-codesign %t_tsan-binary
+// RUN: env %env-TSAN_OPTIONS=abort_on_error=0 %target-run %t_tsan-binary 2>&1 | %FileCheck %s --implicit-check-not='ThreadSanitizer'
+// REQUIRES: executable_test
+// REQUIRES: tsan_runtime
+
+// rdar://101876380
+// UNSUPPORTED: OS=ios
+
+// FIXME: This should be covered by "tsan_runtime"; older versions of Apple OSs
+// don't support TSan.
+// UNSUPPORTED: remote_run
+
+// Test that we do not report a race on initialization; Swift doesn't initialize
+// globals at start-up, but rather uses swift_once().  This is thread safe, but
+// on some platforms TSan wasn't seeing the synchronization, so would report
+// a false positive.
```

Comment:
Please extend this comment to explain how this is testing `swift_once()` (Swift internal!) to avoid confusing, since the test code itself doesn't mention `once()`.

Reply:
Good point. The actual original symptoms of the problem are that code like the code in the test was triggering TSan messages, so from that perspective `swift_once()` is also an implementation detail, but expanding the comment seems good.